src/web/app/user-join-page.component.ts

```diff
@@ -49,39 +49,38 @@ export class UserJoinPageComponent implements OnInit {
         return;
       }
 
-      this.courseService.getJoinCourseStatus(this.key, this.entityType).subscribe((resp: JoinStatus) => {
-        this.hasJoined = resp.hasJoined;
-        this.userId = resp.userId || '';
-        if (this.hasJoined && this.userId) {
-          // The regkey has been used and there is a logged in user.
-          // Simply redirect the user to their home page, regardless of whether the regkey matches or not.
-          this.navigationService.navigateByURL(this.router, `/web/${this.entityType}/home`);
-        } else {
-          this.isLoading = false;
-        }
-      }, (resp: ErrorMessageOutput) => {
-        if (resp.status === 403) {
+      const nextUrl: string = `${window.location.pathname}${window.location.search.replace(/&/g, '%26')}`;
+      this.authService.getAuthUser(undefined, nextUrl).subscribe((auth: AuthInfo) => {
+        if (!auth.user) {
           this.isLoading = false;
-          const nextUrl: string = `${window.location.pathname}${window.location.search.replace(/&/g, '%26')}`;
-          this.authService.getAuthUser(undefined, nextUrl).subscribe((auth: AuthInfo) => {
-            if (!auth.user) {
-              window.location.href = `${this.backendUrl}${auth.studentLoginUrl}`;
+          window.location.href = `${this.backendUrl}${auth.studentLoginUrl}`;
+        } else {
```

Comment:
Instead of using `else` block, you may consider doing this instead:
```ts
  window.location.href = `${this.backendUrl}${auth.studentLoginUrl}`;
  return;
}
// content of else block goes here
```